translib/transformer/subscribe_req_xlate.go

```diff
@@ -0,0 +1,2135 @@
+	NeedCache        bool
+	PType            NotificationType
+	OnChange         OnchangeMode
+	TrgtNodeChld     bool // to indicate the immediate child level pathXlate info of the target node
+	reqLogId         string
+	chldXlateInfos   []*XfmrSubscribePathXlateInfo
+	HandlerFunc      apis.ProcessOnChange // if its true, stop traversing the ygXpathNode's child nodes further
+	IsDataSrcDynamic bool
+}
+
+type XfmrSubscribeReqXlateInfo struct {
+	TrgtPathInfo    *XfmrSubscribePathXlateInfo
+	ChldPathsInfo   []*XfmrSubscribePathXlateInfo
+	ygXpathTrgtInfo *yangXpathInfo
+}
+
+func (reqXlator *subscribeReqXlator) getSubscribePathXlator(gPath *gnmipb.Path, uriPath string, ygXpathInfo *yangXpathInfo,
+	parentXlateInfo *XfmrSubscribePathXlateInfo, xpathYgNode *ygXpathNode) (*subscribePathXlator, error) {
+	var err error
+	reqXlator.pathXlator.gPath = gPath
+	reqXlator.pathXlator.pathXlateInfo = &(XfmrSubscribePathXlateInfo{Path: gPath, ygXpathInfo: ygXpathInfo, reqLogId: reqXlator.subReq.reqLogId})
+	reqXlator.pathXlator.pathXlateInfo.chldXlateInfos = make([]*XfmrSubscribePathXlateInfo, 0)
+	reqXlator.pathXlator.pathXlateInfo.OnChange = OnchangeDefault
+	if ygXpathInfo.subscriptionFlags.Has(subsOnChangeDisable) {
+		reqXlator.pathXlator.pathXlateInfo.OnChange = OnchangeDisable
+	} else if ygXpathInfo.subscriptionFlags.Has(subsOnChangeEnable) {
+		reqXlator.pathXlator.pathXlateInfo.OnChange = OnchangeEnable
+	}
+	reqXlator.pathXlator.uriPath = uriPath
+	reqXlator.pathXlator.keyXfmrYgXpathInfo = ygXpathInfo
+	reqXlator.pathXlator.parentXlateInfo = parentXlateInfo
+	reqXlator.pathXlator.xpathYgNode = xpathYgNode
+	if reqXlator.subReq.xlateNodeType == TARGET_NODE {
+		if err = reqXlator.pathXlator.setTrgtYgXpathInfo(); err != nil {
+			if log.V(dbLgLvl) {
+				log.Warning(reqXlator.subReq.reqLogId, "Error in setting the ygXpathInfo of the last LIST node in the path and the error is :", err)
+			}
+			return nil, err
+		}
+	}
+	return reqXlator.pathXlator, err
+}
+
+func (pathXltr *subscribePathXlator) setTrgtYgXpathInfo() error {
+	if log.V(dbLgLvl) {
+		log.Infof("%v Entering into the setTrgtListYgXpathInfo: ygPath: %v", pathXltr.subReq.reqLogId, pathXltr.subReq.ygPath)
+	}
+
+	ygXpathInfo := pathXltr.pathXlateInfo.ygXpathInfo
+	pathXltr.keyXfmrYgXpathInfo = ygXpathInfo
+	ygPathTmp := pathXltr.subReq.ygPath
+
+	for ygXpathInfo != nil && len(ygXpathInfo.xfmrKey) == 0 {
+		tIdx := strings.LastIndex(ygPathTmp, "/")
+		// -1: not found, and 0: first character in the path
+		if tIdx > 0 {
+			ygPathTmp = ygPathTmp[0:tIdx]
+		} else {
+			break
+		}
+		if log.V(dbLgLvl) {
+			log.Infof("%v xPathTmp: %v", pathXltr.subReq.reqLogId, ygPathTmp)
+		}
+		if ygXpathInfoTmp, ok := xYangSpecMap[ygPathTmp]; !ok || ygXpathInfoTmp == nil {
+			log.Warningf("%v xYangSpecMap does not have the yangXpathInfo for the path: %v", pathXltr.subReq.reqLogId, ygPathTmp)
+			return tlerr.InternalError{Format: "xYangSpecMap does not have the yangXpathInfo", Path: ygPathTmp}
+		} else {
```

Comment:
Consider removing this else block since code returns in the above if block http://google.github.io/styleguide/go/decisions#indent-error-flow